client/rest-high-level/src/main/java/org/elasticsearch/client/indexlifecycle/PhaseExecutionInfo.java

```diff
@@ -0,0 +1,135 @@
+    private static final ParseField MODIFIED_DATE_FIELD = new ParseField("modified_date");
+
+    private static final ConstructingObjectParser<PhaseExecutionInfo, String> PARSER = new ConstructingObjectParser<>(
+        "phase_execution_info", false,
+        (a, name) -> new PhaseExecutionInfo((String) a[0], (Phase) a[1], (long) a[2], (long) a[3]));
+    static {
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), POLICY_NAME_FIELD);
+        PARSER.declareObject(ConstructingObjectParser.constructorArg(), Phase::parse, PHASE_DEFINITION_FIELD);
+        PARSER.declareLong(ConstructingObjectParser.constructorArg(), VERSION_FIELD);
+        PARSER.declareLong(ConstructingObjectParser.constructorArg(), MODIFIED_DATE_FIELD);
+    }
+
+    public static PhaseExecutionInfo parse(XContentParser parser, String name) {
+        return PARSER.apply(parser, name);
+    }
+
+    private final String policyName;
+    private final Phase phase;
+    private final long version;
+    private final long modifiedDate;
+
+    /**
+     * This class holds information about the current phase that is being executed
+     * @param policyName the name of the policy being executed, this may not be the current policy assigned to an index
+     * @param phase the current phase definition executed
+     * @param version the version of the <code>policyName</code> being executed
+     * @param modifiedDate the time the executing version of the phase was modified
+     */
+    public PhaseExecutionInfo(String policyName, Phase phase, long version, long modifiedDate) {
+        this.policyName = policyName;
+        this.phase = phase;
+        this.version = version;
+        this.modifiedDate = modifiedDate;
+    }
+
+    public String getPolicyName() {
+        return policyName;
+    }
+
+    public Phase getPhase() {
+        return phase;
+    }
+
+    public long getVersion() {
+        return version;
+    }
+
+    public long getModifiedDate() {
+        return modifiedDate;
+    }
+
+    public String getModifiedDateString() {
+        ZonedDateTime modifiedDateTime = ZonedDateTime.ofInstant(Instant.ofEpochMilli(modifiedDate), ZoneOffset.UTC);
+        return modifiedDateTime.toString();
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(policyName, phase, version, modifiedDate);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        PhaseExecutionInfo other = (PhaseExecutionInfo) obj;
+        return Objects.equals(policyName, other.policyName) &&
+            Objects.equals(phase, other.phase) &&
+            Objects.equals(version, other.version) &&
+            Objects.equals(modifiedDate, other.modifiedDate);
+    }
+
+    @Override
+    public String toString() {
+        return Strings.toString(this, true, true);
```

Comment:
Can you make this non-pretty, it's always weird when you log things and then end up being multi-line

Reply:
true, log lines are unfortunate, but comparing test assertions are. since one is in production, logging concerns win 👍